Modules/IO/JPEG/src/itkJPEGImageIO.cxx

```diff
@@ -364,8 +364,8 @@ JPEGImageIO::ReadImageInformation()
 
   // jpeg_calc_output_dimensions to calculate cinfo.output_components
   jpeg_calc_output_dimensions(&cinfo);
-  if constexpr (sizeof(void *) < 8 && (static_cast<unsigned long long>(cinfo.output_width) * cinfo.output_height *
```

Comment:
Which compiler complains about this?

Reply:
I think any compiler on x86 (32 bit) will complain. On x86_64 the second part of the expression (cinfo) is unreachable.

Reply:
- Interesting... the `constexpr` was added by pull request #5086 commit e1e2cd10d926c1bc366edf3fd29c2f95f47a9d83 (by Hans, @hjmjohnson), on Dec 18, 2024. So apparently 32-bit compilation is broken for a while already!

A `constexpr`-preserving alternative could be:

```cpp
  if constexpr (sizeof(void *) < 8)
  {
    if ((static_cast<unsigned long long>(cinfo.output_width) * cinfo.output_height * cinfo.output_components) >
        0xffffffff)
    {
      jpeg_destroy_decompress(&cinfo);
      itkExceptionMacro("JPEG image is too big " << this->GetFileName());
    }
  }
```

But that's just a minor suggestion 🤷  Thanks @WangWeiLin-MV !

My two cents, Niels